easybuild/easyblocks/l/llvmcore.py

```diff
@@ -0,0 +1,748 @@
+#   -DCMAKE_VERBOSE_MAKEFILE=1
+#   -DCPACK_GENERATOR=DEB
+#   ‘-DCMAKE_INSTALL_RPATH=$ORIGIN:$ORIGIN/…/lib:$ORIGIN/…/lib64:/opt/rocm-5.6.0-9999/lib:/opt/rocm-5.6.0-9999/lib64:/opt/rocm/lib:/opt/rocm/lib64:$ORIGIN/…/llvm/lib’
+#   -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=FALSE
+#   -DROCM_PATCH_VERSION=50600
+#   -DCMAKE_INSTALL_PREFIX=/home/ampandey/rocm-toolchain/rocm-5.6
+#   -DCPACK_PACKAGING_INSTALL_PREFIX=/home/ampandey/rocm-toolchain/rocm-5.6
+#   -DCMAKE_INSTALL_PREFIX=/home/ampandey/rocm-toolchain/rocm-5.6/llvm
+#   ‘-DLLVM_TARGETS_TO_BUILD=AMDGPU;X86’
+#   ‘-DLLVM_ENABLE_PROJECTS=clang;lld;clang-tools-extra’
+#   ‘-DLLVM_ENABLE_RUNTIMES=libcxx;libcxxabi;compiler-rt’
+#   -DLIBCXX_ENABLE_SHARED=OFF
+#   -DLIBCXX_ENABLE_STATIC=ON
+#   -DLIBCXX_INSTALL_LIBRARY=OFF
+#   -DLIBCXX_INSTALL_HEADERS=OFF
+#   -DLIBCXXABI_ENABLE_SHARED=OFF
+#   -DLIBCXXABI_ENABLE_STATIC=ON
+#   -DLIBCXXABI_INSTALL_STATIC_LIBRARY=OFF
+#   -DLLVM_BUILD_DOCS=OFF
+#   -DLLVM_ENABLE_SPHINX=OFF
+#   -DSPHINX_WARNINGS_AS_ERRORS=OFF
+#   -DSPHINX_OUTPUT_MAN=OFF
+#   -DLLVM_ENABLE_ASSERTIONS=1
+#   -DLLVM_ENABLE_Z3_SOLVER=OFF
+#   -DLLVM_ENABLE_ZLIB=ON
+#   -DLLVM_AMDGPU_ALLOW_NPI_TARGETS=ON
+#   -DCLANG_DEFAULT_PIE_ON_LINUX=0
+#   -DCLANG_DEFAULT_LINKER=lld
+#   -DCLANG_DEFAULT_RTLIB=compiler-rt
+#   -DCLANG_DEFAULT_UNWINDLIB=libgcc
+#   -DPACKAGE_VENDOR=AMD
+#   -DLLVM_BUILD_LLVM_DYLIB=OFF
+#   -DLLVM_LINK_LLVM_DYLIB=OFF
+#   -DLLVM_ENABLE_LIBCXX=OFF
+#   /home/ampandey/rocm-toolchain/build/…/external/llvm-project/llvm
+
+
+# https://github.com/llvm/llvm-project/issues/72108
+# -DCMAKE_BUILD_TYPE="Release" \
+# -DCMAKE_C_FLAGS="-Wno-backend-plugin" \
+# -DCMAKE_CXX_FLAGS="-Wno-backend-plugin" \
+# -DCMAKE_C_COMPILER="clang" \
+# -DCMAKE_CXX_COMPILER="clang++" \
+# -DLLVM_ENABLE_CLASSIC_FLANG=ON \
+# -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;lld;openmp" \
+# -DLLVM_ENABLE_RUNTIMES="compiler-rt;libcxx;libcxxabi" \
+# -DLLVM_LIBDIR_SUFFIX=64 \
+# -DLLVM_USE_LINKER=lld \
+# -DLLVM_PARALLEL_COMPILE_JOBS="96" \
+# -DLLVM_PARALLEL_LINK_JOBS=96 \
+# -DLLVM_ENABLE_NEW_PASS_MANAGER=ON \
+# -DLLVM_TARGETS_TO_BUILD="X86;BPF;WebAssembly" \
+# -DLLVM_ENABLE_LIBCXX=OFF \
+# -DLLVM_STATIC_LINK_CXX_STDLIB=OFF \
+# -DLLVM_BINUTILS_INCDIR="/usr/include" \
+# -DLLVM_ENABLE_TERMINFO=OFF \
+# -DLLVM_ENABLE_LIBXML2=OFF \
+# -DLLVM_ENABLE_LIBEDIT=OFF \
+# -DLLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF \
+# -DCOMPILER_RT_BUILD_LIBFUZZER=OFF \
+# -DCOMPILER_RT_BUILD_XRAY=OFF \
+# -DCOMPILER_RT_BUILD_ORC=OFF \
+# -DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON \
+# -DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=OFF \
+# -DOPENMP_ENABLE_LIBOMPTARGET=OFF \
+# -DOPENMP_ENABLE_OMPT_TOOLS=OFF \
+# -DOPENMP_ENABLE_TESTING=OFF \
+# -DCOMPILER_RT_CXX_LIBRARY=libcxx \
+# -DSANITIZER_CXX_ABI=libc++ \
+# -DSANITIZER_CXX_ABI_INTREE=ON \
+# -DSANITIZER_USE_STATIC_CXX_ABI=ON \
+# -DSANITIZER_TEST_CXX=libc++ \
+# -DSANITIZER_TEST_CXX_INTREE=ON \
+# -DSANITIZER_USE_STATIC_TEST_CXX=ON \
+# -DSANITIZER_LIT_USE_LIBCXX=ON \
```

Comment:
Is there any reason for the comments being present? It looks like some are based on LLVM issues or other discussions.

Reply:
No, they were leftovers to keep track of some of the combinations i had tried.
Since now also the non bootstrapped build is working I've removed them in https://github.com/easybuilders/easybuild-easyblocks/pull/3373/commits/541e65a73973fc4438dbb0378a2e3e2a359225bd.

Also thanks for the fast review and sorry for the late reply i went on holiday right after opening the PR.